packages/@react-aria/dnd/docs/useDraggableCollection.mdx

```diff
@@ -0,0 +1,675 @@
+  let dropState = useDroppableCollectionState({
+    ...props,
+    collection: state.collection,
+    selectionManager: state.selectionManager
+  });
+
+  let {collectionProps} = useDroppableCollection({
+    ...props,
+    // Provide drop targets for keyboard and pointer-based drag and drop.
+    keyboardDelegate: new ListKeyboardDelegate(state.collection, state.disabledKeys, ref),
+    dropTargetDelegate: new ListDropTargetDelegate(state.collection, ref)
+  }, dropState, ref);
+  ///- end highlight -///
+
+  return (
+    /*- begin highlight -*/
+    <ul
+      {...mergeProps(listBoxProps, collectionProps)}
+      /*- end highlight -*/
+      ref={ref}>
+      {[...state.collection].map((item) => (
+        <ReorderableOption
+          key={item.key}
+          item={item}
+          state={state}
+          dragState={dragState}
+          /*- begin highlight -*/
+          dropState={dropState}
+          /*- end highlight -*/
+        />
+      ))}
+    </ul>
+  );
+}
+
+function ReorderableOption({ item, state, dragState, dropState }) {
+  ///- begin collapse -///
+  let ref = React.useRef();
+  let { optionProps } = useOption({key: item.key}, state, ref);
+  let { isFocusVisible, focusProps } = useFocusRing();
+
+  // Register the item as a drag source.
+  let {dragProps} = useDraggableItem({
+    key: item.key
+  }, dragState);
+  ///- end collapse -///
+  // ...
+
+  return (
+    <>
+      {/*- begin highlight -*/}
+      <DropIndicator
+        target={{type: 'item', key: item.key, dropPosition: 'before'}}
+        dropState={dropState} />
+      {/*- end highlight -*/}
+      <li
+        {...mergeProps(optionProps, dragProps, focusProps)}
+        ref={ref}
+        className={`option ${isFocusVisible ? 'focus-visible' : ''}`}>
+        {item.rendered}
+      </li>
+      {/*- begin highlight -*/}
+      {state.collection.getKeyAfter(item.key) == null &&
+        <DropIndicator
+          target={{type: 'item', key: item.key, dropPosition: 'after'}}
+          dropState={dropState} />
+      }
+      {/*- end highlight -*/}
+    </>
+  );
+}
+
+///- begin highlight -///
+function DropIndicator(props) {
+  let ref = React.useRef();
+  let {dropIndicatorProps, isHidden, isDropTarget} = useDropIndicator(props, props.dropState, ref);
+  if (isHidden) {
+    return null;
+  }
+
+  let className = props.target.type === 'item'
+    ? `drop-indicator ${isDropTarget ? 'drop-target' : ''}`
+    : '';
+
+  return (
+    <li
+      {...dropIndicatorProps}
+      role="option"
+      ref={ref}
+      className={className} />
+  );
+}
+///- end highlight -///
+```
+
+Now, we can render an example ListBox, which allows the user to reorder items. The `onReorder` event is triggered when the user drops dragged items which originated within the same collection. As above, [useListData](../react-stately/useListData.html) is used to manage the list items in this example, but it is not a requirement.
+
+```tsx example
```

Comment:
Just noticed this example actually doesn't exit keyboard dragging mode when you confirm your drop via Enter. Also  if you select the top three values and attempt to drop them between the remaining two items they are appended to the end of the list instead.

Reply:
Good catch. The keyboard issue should be fixed by the NVDA PR that just went in. The other issue with reordering seems like maybe something with useListData. I'd prefer to look at that separately.